### ISSUE TITLE
2316632087 (0x8A150017) Error when using `winget install`

### ISSUE THREAD

**Issue by KK-Designs** [microsoft/winget-cli#2808]
I am getting the error code on exit, `2316632087 (0x8A150017)`, which translates to [`APPINSTALLER_CLI_ERROR_NO_MANIFEST_FOUND`](https://github.com/microsoft/winget-cli/blob/master/src/AppInstallerCommonCore/Public/AppInstallerErrors.h#:~:text=0x8A150017 "Error code name") when running the command `winget install --id=Discord.Discord -v "1.0.9004" -e --accept-package-agreements --accept-source-agreements` in nodejs. Running it in a UI shell seems to work fine.

My code:

```js
const cmd = require('child_process').spawn;
const cmdStream = cmd('winget', ['install', `--id=Discord.Discord`, '-v', `"1.0.9004"`, '-e', '--accept-package-agreements', '--accept-source-agreements']);
const updateAppData = [];
let updatePrep = false;
let updateMsg = true;

cmdStream.stdout.on('data', (data) => {
	data = data.toString();
	console.log(data);
});

cmdStream.stderr.on('data', (data) => {
	console.log('stderr: ' + data.toString());
});

cmdStream.on('exit', async (code) => {
	console.log('child process exited with code ' + code.toString());
});
```

Why does this happen?

**Comment by R-Adrian** [comment]
is this on Windows 10? using TLS 1.3?
if so, can you please try using TLS 1.2 instead of letting the script to auto-select TLS 1.3 ?

note: there's a pinned issue about the error with 0 kb manifest files #2718

**Comment by KK-Designs** [comment]
> is this on Windows 10? Using TLS 1.3? Please try using TLS 1.2 instead of letting the script auto-select TLS 1.3.
> 
> note: there's a pinned issue about the error with 0 kb manifest files #2718

I am on windows 11; I do not know how to check my current TLS version. Could you tell me how?

**Comment by stephengillie** [comment]
Which version of Node.js is in use? 

Could you append `--verbose-logs` to your command and run again? 

Winget logs can be found here, or at the location from `winget --info`:
> %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\DiagOutputDir

**Comment by R-Adrian** [comment]
> I am on windows 11; I do not know how to check my current TLS version. Could you tell me how?
If you have not modified them since installation then Win11 will default to have support  for v1.3 enabled.
Windows 10 does not have v1.3 (has just experimental support, not enabled by default).

**https://learn.microsoft.com/en-us/windows/win32/secauthn/tls-cipher-suites-in-windows-11**

This can be configured in a few places: (wiki with pictures on MS Technet below)
https://social.technet.microsoft.com/wiki/contents/articles/54255.troubleshooting-turn-on-tls-1-0-tls-1-1-and-tls-1-2-in-advanced-settings.aspx
note: that article is a bit old but is still valid, just do not touch other versions of TLS and especially make sure not to modify anything for TLS 1.2 just for v1.3.

TL;DR locations summary:
a) system-wide in the registry under
[HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\SecurityProviders\SCHANNEL\Protocols\TLS 1.3\Client]
-needs a reboot after changing settings here
-it is OK for this configuration key to be missing - it just means that the OS uses default settings (same for other subkeys under the Protocols)

b) for each user account individually by using the Advanced section of the Internet Options control panel, (start->run->inetcpl.cpl)

**Comment by KK-Designs** [comment]
I have found and fixed my issue. When I checked the logs it said that the version could not be found, and that's because the command that was getting executed from my code sent the version as `""1.0.9004""` which double-double quotes. Removing the extra quotes worked.

Code:
```js
const cmd = require('child_process').spawn;
const cmdStream = cmd('winget', ['install', `--id=Discord.Discord`, '-v', `1.0.9004`, '-e', '--accept-package-agreements', '--accept-source-agreements']);
const updateAppData = [];
let updatePrep = false;
let updateMsg = true;

cmdStream.stdout.on('data', (data) => {
	data = data.toString();
	console.log(data);
});

cmdStream.stderr.on('data', (data) => {
	console.log('stderr: ' + data.toString());
});

cmdStream.on('exit', async (code) => {
	console.log('child process exited with code ' + code.toString());
});
```
